doc/dev/developer_guide/basic-workflow.rst

```diff
@@ -229,51 +332,53 @@ Understanding Automated PR validation
+result is shown on GitHub in the pull request itself.
 
-You should test your modifications before you open a PR.
-Refer to the chapters on testing for details.
+Test your modifications before you open a PR.  Refer to the chapters
+on testing for details.
 
 Notes on PR make check test
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The GitHub :ref:`make check<make-check>` test is driven by a Jenkins instance.
 
 Jenkins merges your PR branch into the latest version of the base branch before
-starting tests. This means that you don't have to rebase the PR to pick up any fixes.
+it starts any tests. This means that you don't have to rebase the PR in order
+to pick up any fixes.
 
 You can trigger PR tests at any time by adding a comment to the PR - the
-comment should contain the string "test this please". Since a human subscribed
-to the PR might interpret that as a request for him or her to test the PR, we
-recommend that you address Jenkins directly. For example, write "jenkins retest
-this please".  For efficiency a single re-test can also be requested with
-e.g. "jenkins test signed".  For reference, a list of these requests is
-automatically added to the end of each new PR's description.
+comment should contain the string "test this please". Since a human who is
+subscribed to the PR might interpret that as a request for him or her to test
+the PR, you must address Jenkins directly. For example, write "jenkins retest
+this please". If you need to run only one of the tests, you can request it with
+a command like "jenkins test signed". A list of these requests is automatically
+added to the end of each new PR's description, so check there to find the
+single test you need.
 
 If there is a build failure and you aren't sure what caused it, check the
-:ref:`make check<make-check>` log. To access it, click on the "details" (next
-to the :ref:`make check<make-check>` test in the PR) link to enter the Jenkins web
-GUI. Then click on "Console Output" (on the left).
-
-Jenkins is configured to search logs for strings known to have been associated
-with :ref:`make check<make-check>` failures in the past. However, there is no
-guarantee that these known strings are associated with any given
-:ref:`make check<make-check>` failure. You'll have to read through the log to determine the
-cause of your specific failure.
+:ref:`make check<make-check>` log. To access the make check log, click the
+"details" (next to the :ref:`make check<make-check>` test in the PR) link to
+enter the Jenkins web GUI. Then click "Console Output" (on the left).
+
+Jenkins is configured to search logs for strings that are known to have been
+associated with :ref:`make check<make-check>` failures in the past. However,
+there is no guarantee that these known strings are associated with any given
+:ref:`make check<make-check>` failure. You'll have to read through the log to
+determine the cause of your specific failure.
 
```

Comment:
I think it might be valuable to include a brief section here about refinements / required changes from feedback or test errors, with an example of amending the commit and force-pushing, so nobody has to squash.

Reply:
I agree with you even before you said it. I have a stack of 4x6 cards on my desk that contain intended improvements to this file, and those improvements include a step-by-step handholder through the process of amending, interactively rebasing (squashing) and force-pushing.

However, (as you can see) this is already a very long change to a fundamental document. In the interest of getting some changes into this document rather than continuing to refine and refine until probably July, I elected to go ahead and make a PR.

I envision the page eventually divided into two sections: 1. a step-by-step guide to making a PR, and 2. reference material after the step-by-step guide that can be used to deepen you understanding of the steps in the step-by-step guide.